lib/rules/no-unused-prop-types.js

```diff
@@ -893,7 +454,7 @@ module.exports = {
         return;
       }
 
-      (props || []).forEach(prop => {
+      Object.values(props || {}).forEach(prop => {
```

Comment:
`declaredPropTypes` used to be an array here, but an object in `prop-types`. Common implementation uses an object (keyed by prop name) for easier lookups